Source/ChocolateyGui.Common/Startup/AssemblyResolver.cs

```diff
@@ -0,0 +1,121 @@
+﻿// <copyright file="AssemblyResolver.cs" company="Chocolatey">
+// Copyright 2017 - Present Chocolatey Software, LLC
+// Copyright 2014 - 2017 Rob Reynolds, the maintainers of Chocolatey, and RealDimensions Software, LLC
+// </copyright>
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using Serilog;
+
+namespace ChocolateyGui.Common.Startup
+{
+    public class AssemblyResolver
+    {
+        private const int LOCKRESOLUTIONTIMEOUTSECONDS = 5;
+        private static readonly object _lockObject = new object();
+
+        public static Assembly ResolveOrReloadChocolateyAssembly(string assemblyFileLocation)
+        {
+            return ResolveOrLoadAssemblyInternal(
+                "chocolatey",
+                string.Empty,
+                assemblyFileLocation,
+                (assembly) =>
+                {
+                    var assemblyName = assembly.GetName();
+
+                    return string.Equals(assemblyName.Name, "chocolatey", StringComparison.OrdinalIgnoreCase) &&
+                        assemblyName.Version != new Version(0, 10, 15, 0); // TODO: Should maybe be created automatically

```

Comment:
Not sure what you mean here by this TODO. Can you elaborate?

Reply:
The initial intention was that it should only try to load the `choco.exe` if it was not the version we already referenced, however, this is leftover from the initial testing and I need to remove this function completely.